libraries/botbuilder-azure/src/cosmosDbKeyEscape.ts

```diff
@@ -48,10 +53,14 @@ export namespace CosmosDbKeyEscape {
             ''
         );
 
-        return truncateKey(sanitizedKey);
+        return truncateKey(`${sanitizedKey}${keySuffix || ''}`, compatibilityMode);
     }
 
-    function truncateKey(key: string): string {
+    function truncateKey(key: string, truncateKeysForCompatibility?: boolean): string {
+        if (truncateKeysForCompatibility === false) {
+            return key;
```

Comment:
I'd be surprised if it ever exceeded it, but should we also do a check to ensure `key.length < 1024`? For BotState storage, this will probably never occur, but I could see somebody maybe doing something weird like stringifying some JSON and using that as a key for custom storage.

Reply:
That was considered, but a decision was made to to let cosmosdb throw an exception under this condition.